### ISSUE TITLE
Get action "chemical_gene_regulation_graph_expansion" into ARAX_infer

### ISSUE THREAD

**Issue by chunyuma** [RTXteam/RTX#1961]
This issue summarizes the steps to integrate action `chemical_gene_regulation_graph_expansion` into ARAX_infer:

- [ ] Train two models (one for predicting `increase gene activity while the other for `decrease ...`)
- [ ] implement prediction in Infer folder (add a new script `creativeCRG.py`) Note: `CRG` means "Chemical regulates gene".
- [ ] Modify `ARAX_infer`, `infer_utilities.py`, `RTX_Configuration.py`, `config_dbs.json`, `ARAX_expander.py` and `ARAX_database_manager.py`
- [ ] Upload the model files to `arax-databases.rtx.ai` server
- [ ] Update the [Document](https://github.com/RTXteam/RTX/tree/master/code/ARAX/Documentation)
- [ ] Do unit tests
- [ ] Download transcription factor list and consider the transcription factor for path extraction
- [ ] Do pre-computation for increasing speed

**Comment by chunyuma** [comment]
The action "chemical_gene_regulation_graph_expansion" has been added into ARAX_infer. Please see the instruction in the [Document](https://github.com/RTXteam/RTX/tree/master/code/ARAX/Documentation). I also added [a few tests](https://github.com/RTXteam/RTX/blob/master/code/ARAX/test/test_ARAX_infer.py#L188-L335) in test suites. But some of them have an error described below:

@amykglen and @edeutsch, I tried to add `qualifiers` in the query graph but got an error: `ERROR: [UnknownQEdgeProperty] QueryGraph edge 'r_edge' has an unexpected property 'qualifiers'. This property is not understood and therefore processing is halted, rather than answer an incompletely understood query.` [Here](https://github.com/RTXteam/RTX/blob/master/code/ARAX/test/test_ARAX_infer.py#L206-L248) is an example. It seems like the query graph currently doesn't allow to have `qualifiers`.

**Comment by amykglen** [comment]
yes, query graphs have `qualifier_constraints`, not `qualifiers`. see the TRAPI documentation: 

https://github.com/NCATSTranslator/ReasonerAPI/blob/master/docs/reference.md#qedge-

**Comment by chunyuma** [comment]
Thanks @amykglen. Now all `chemical_gene_regulation_graph_expansion ` tests pass.

One more question: What is the function of `.filled` method in the[ query_graphedges[key]](https://github.com/RTXteam/RTX/blob/e5cee5c643f987ddcd19b6d24dd165bc8a5c9d32/code/ARAX/ARAXQuery/Infer/scripts/infer_utilities.py#L235) ?

**Comment by amykglen** [comment]
sorry @chunyuma, I missed your question back in February! `QEdge.filled` is just a variable that we use within ARAX to signal to Resultify whether a given QEdge has already been processed/expanded. This is needed because otherwise Resultify can't tell whether a given QEdge simply had no answers vs. answers haven't even been looked for, and thus allows it to resultify partially-expanded queries (as needed in #1848)

**Comment by edeutsch** [comment]
@chunyuma what is the status on this item?

**Comment by dkoslicki** [comment]
Triage from 6/28 AHM: focus first on DrugConflator & g() function before going after this one. Recall remaining TODOs are:

- [ ] make explanatory paths go through miRNAs or transcription factors
- [ ] Check that the combination of `affects` directions makes sense (eg. if "upregulate" the "up regulate something that upregulates" is acceptable, but not "upregulate something that downregulates")
- [ ] Possibly pre-compute for speed

**Comment by dkoslicki** [comment]
With current xCRG implementation, we won't be precomputing everything, so this can be closed.
